Investigations/MIMIC_Notebook.ipynb

```diff
@@ -0,0 +1,531 @@
+    "\n",
+    "plot_variable_distributions(\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    data_supp=data_supp,\n",
+    "    synthetic_supp=synthetic_supp,\n",
+    "    saving_filepath=\"\",\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "90f1ebb0",
+   "metadata": {},
+   "source": [
+    "## Metric evaluation\n",
+    "\n",
+    "For datasets that have datetime columns, we need to re-transform these into a numerical value as our metrics cannot handle datetime objects. We are then inputting <b>original_metric_set</b> alongside the newly transformed synthetic set i.e. <b>metric_synthetic_supp</b>. If datetimes are not included in the set then you can just run <b>data_supp</b> against <b>synthetic_supp</b> and skip the datetime handling.\n",
+    "\n",
+    "We use the SDV evaluation framework. Supply the metrics you wish to find in the user_metrics list from SDV guidance. Can start here: https://sdv.dev/SDV/user_guides/evaluation/single_table_metrics.html\n",
+    "\n",
+    "Note that not all of these will work, some are hit and miss. We predominantly rely on continuous and discrete KL divergence measures. You can also input <b>\"gower\"</b> and this will calculate the gower distance using the gower library."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d845b202",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- Datetime Handling -------- #\n",
+    "\n",
+    "# If the dataset has datetimes then we need to re-convert these to a numerical\n",
+    "# Value representing seconds, this is so we can evaluate the metrics on them\n",
+    "\n",
+    "metric_synthetic_supp = synthetic_supp.copy()\n",
+    "\n",
+    "for index, column in enumerate(original_datetime_columns):\n",
+    "\n",
+    "    # Fit datetime transformer - converts to seconds\n",
+    "    temp_datetime = datetime.DatetimeTransformer()\n",
+    "    temp_datetime.fit(metric_synthetic_supp, columns=column)\n",
+    "\n",
+    "    metric_synthetic_supp = temp_datetime.transform(metric_synthetic_supp)\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f7e4174b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#%% -------- SDV Metrics -------- #\n",
+    "\n",
+    "# Define the metrics you want the model to evaluate\n",
+    "\n",
+    "gower = False\n",
+    "\n",
+    "metrics = distribution_metrics(\n",
+    "    gower_bool=gower,\n",
+    "    data_supp=original_metric_set,\n",
+    "    synthetic_supp=metric_synthetic_supp,\n",
+    "    categorical_columns=original_categorical_columns,\n",
+    "    continuous_columns=original_continuous_columns,\n",
+    "    saving_filepath=\"\",\n",
+    "    pre_proc_method=pre_proc_method,\n",
+    ")\n"
+   ]
```

Comment:
Possibly add a print of the metrics here to show the outputs in the notebook as well